src/systems/world-update.js

```diff
@@ -0,0 +1,192 @@
+          !this.scale.equals(oneScale) ||
+          !this.matrix.equals(identity)
+        ) {
+          // Only update the matrix the first time if its non-identity, this way
+          // this.matrixIsModified will remain false until the default
+          // identity matrix is updated.
+          this.updateMatrix();
+        }
+
+        this.hasHadFirstMatrixUpdate = true;
+        this.cachedMatrixWorld = this.matrixWorld;
+      } else if (this.matrixNeedsUpdate || this.matrixAutoUpdate || forceLocalUpdate) {
+        this.updateMatrix();
+        if (this.matrixNeedsUpdate) this.matrixNeedsUpdate = false;
+      }
+
+      if (!skipParents && this.parent) {
+        this.parent.updateMatrices(false, forceWorldUpdate, false);
+      }
+
+      if (this.matrixWorldNeedsUpdate || forceWorldUpdate) {
+        if (this.parent === null) {
+          this.matrixWorld.copy(this.matrix);
+        } else {
+          // If the matrix is unmodified, it is the identity matrix,
+          // and hence we can use the parent's world matrix directly.
+          //
+          // Note this assumes all callers will either not pass skipParents=true
+          // *or* will update the parent themselves beforehand as is done in
+          // updateMatrixWorld.
+          if (!this.matrixIsModified) {
+            this.matrixWorld = this.parent.matrixWorld;
+          } else {
+            // Once matrixIsModified === true, this.matrixWorld has been updated to be a local
+            // copy, not a reference to this.parent.matrixWorld (see updateMatrix/applyMatrix)
+            this.matrixWorld.multiplyMatrices(this.parent.matrixWorld, this.matrix);
+          }
+        }
+
+        this.matrixWorldNeedsUpdate = false;
+
+        return true;
+      }
+
+      return false;
+    };
+
+    // Computes this object's matrices and then the recursively computes the matrices of all the children.
+    //
+    // forceWorldUpdate - If true and the object is visible, will force the world matrix to be updated for
+    // this node and all of its children.
+    //
+    // includeInvisible - If true, does not ignore non-visible objects.
+    THREE.Object3D.prototype.updateMatrixWorld = function(forceWorldUpdate, includeInvisible) {
+      if (!this.visible && !includeInvisible) return;
+
+      // Do not recurse upwards, since this is recursing downwards
+      const forceChildrenWorldUpdate = this.updateMatrices(false, forceWorldUpdate, true);
+
+      const children = this.children;
+
+      for (let i = 0, l = children.length; i < l; i++) {
+        children[i].updateMatrixWorld(forceChildrenWorldUpdate, includeInvisible);
+      }
+    };
+  },
+
+  _patchRenderFunc: function() {
+    const renderer = this.el.renderer;
+    const render = renderer.render;
+
+    renderer.render = (scene, camera, renderTarget) => {
+      scene.updateMatrixWorld();
+      render.call(renderer, scene, camera, renderTarget);
+      this.frame++;
```

Comment:
We don't seem to use this anywhere. Just for debugging? The camera and any post effects are probably going to mess this count up.

Reply:
this was incidentally very very useful for debugging since I could measure the updateMatrixWorld cost as part of the main tick, but we can probably take this out if we want